### PR TITLE
DROOLS-1199 Memory leak in KieScanner

### PR DESCRIPTION
**Pull request by baldimir**
- Added test case for KieScanner memory leak issue.

### CODE REVIEW

File: drools-test-coverage/test-suite/src/test/java/org/drools/testcoverage/memory/KieScannerMemoryTest.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+                logger.debug("Average memory: " + averageMemory);
+                averageMemoryFootprints.add(averageMemory);
+                averageMemory = 0;
+            }
+        }
+    }
+
+    private void waitForMillis(final long millis, final long startTimeMillis) {
+        while ((System.currentTimeMillis() - startTimeMillis) < millis) {
+            // do nothing - wait
```
Comment (winklerm) [inline drools-test-coverage/test-suite/src/test/java/org/drools/testcoverage/memory/KieScannerMemoryTest.java:127]: At first, this looked suspicious to me (will eat up one CPU core) but I can see you are waiting for 30 milliseconds at most, so I think it is fine in this case.

Comment (baldimir) [inline drools-test-coverage/test-suite/src/test/java/org/drools/testcoverage/memory/KieScannerMemoryTest.java:127]: Yes it was intentional. I sacrificed CPU for accuracy in this case.

File: drools-test-coverage/test-suite/src/test/java/org/drools/testcoverage/memory/KieScannerMemoryTest.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        }
+    }
+
+    private String getMemoryMeasurmentsString(final List<Long> memoryMeasurements) {
```
Comment (winklerm) [inline drools-test-coverage/test-suite/src/test/java/org/drools/testcoverage/memory/KieScannerMemoryTest.java]: There's a typo in method name - "Measurments".

Comment (baldimir) [inline drools-test-coverage/test-suite/src/test/java/org/drools/testcoverage/memory/KieScannerMemoryTest.java]: Fixed.

**Comment by winklerm** [comment]
Looks good to me, +1 for merging.

**Pull request merged**